### PR TITLE
feat(DB/SAI): lunar festival invitation teleport

### PR DESCRIPTION
**Pull request by Helias**
## Changes Proposed:
-  ported from https://github.com/TrinityCore/TrinityCore/commit/82f6d482732e621882fa536ee04368034f78e00a
- I changed the guid to be more coherent with ours, moreover the guid proposed from the TC PR were already used in our db
image

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4348
- Closes https://github.com/chromiecraft/chromiecraft/issues/19

## Tests Performed:
Tested in-game, I used the item in Ironforge and in Moonglade near humans, trolls, and gnomes.
Of course, if you are alliance and try to teleport to a horde capital it will not work.

## How to Test the Changes:
- create a dwarf or gnome
- `.go xyz -4662.059316 -956.693726 500.893890 0`
- `.additem 21711`
-  use the item
- it will teleport you to Moonglade
- now you can use the item inside the moonlight area near the NPCs of your faction to teleport you to a capital city
- if you use it with NPCs of the opposite faction it will not teleport you

## Known Issues and TODO List:
I saw that in Darnassus there is no moonlight, only the elf that gives you the quest, I don't know if it is normal or not.
And the teleport does not work

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Kitzunu** [comment]
Edited your initial message so it will actually close the chromiecraft one @Helias

**Comment by LannyE** [comment]
I was skulking around the trinitycore forums last night when I had this issue and here is some info on the Darnassus game objects that are missing from the Lunar event [ https://github.com/TrinityCore/TrinityCore/issues/4945]

They claimed this fixed the missing game objects in Darnassus.

-- All ThX & Credits to Darkman1983
INSERT INTO `gameobject` (`guid`,`id`,`map`,`spawnMask`,`phaseMask`,`position_x`,`position_y`,`position_z`,`orientation`,`rotation0`,`rotation1`,`rotation2`,`rotation3`,`spawntimesecs`,`animprogress`,`state`)
VALUES (200000, 300058, 1, 0, 0, 1983.464966, -4255.527344, 31.897110, 0.526572, 0, 0, 0, 0, 300, 0, 1),
(200001, 300058, 1, 0, 0, -1031.407349, -230.353455, 160.180954, 3.209599, 0, 0, 0, 0, 300, 0, 1),
(200002, 300058, 1, 0, 0, 10150.500000, 2602.330078, 1330.829956, 4.751910, 0, 0, 0, 0, 300, 0, 1),
(200003, 300058, 0, 0, 0, 1642.353760, 239.186249, 62.591576, 5.267044, 0, 0, 0, 0, 300, 0, 1),
(200004, 300058, 0, 0, 0, -4663.466797, -955.900879, 500.377686, 5.617465, 0, 0, 0, 0, 300, 0, 1),
(200005, 300058, 0, 0, 0, -8747.168945, 1073.401367, 90.419289, 5.551456, 0, 0, 0, 0, 300, 0, 1);

INSERT INTO `game_event_gameobject` (`eventEntry`, `guid`)
VALUES (7, 200000),
(7, 200001),
(7, 200002),
(7, 200003),
(7, 200004),
(7, 200005);

**Comment by Helias** [comment]
@LannyE, next time use this markdown properties
image

it will show you the code as now it is, I edited your post above :wink:

**Comment by LannyE** [comment]
Thanks for showing me how to do it.  I spent about 5 minutes trying to get the insert code selection to work before I gave up.  I attempted the SQL but realized that gameopbject 300058 doesn't exist.  I alse noticed that the Azerothcore database only shows 5 Lunar Festival Harbingers creatures.  In classic Wow there should be 6.  One for each capitol.  In WOTLK, I think, there should also be one in Shattrath and Dalaran [https://www.wowhead.com/npc=15895/lunar-festival-harbinger]()  Darnassus should have a Lunar Festival Harbinger including the moonlight and launchers situated on the tree trunk above Elder Bladeswift.

**Comment by Kitzunu** [comment]
Import the gameobject.

File: data/sql/updates/pending_db_world/rev_1611685103659442481.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+(22,3,180867,1,0,29,0,15906,5,0,0,0,'','Lunar FestivalExecute sai if near npc 15906'),
+(22,3,180867,1,0,6,0,469,0,0,0,0,'','Lunar Festival SAI the player Must Be Alliance'),
+(22,4,180867,1,0,29,0,15905,5,0,0,0,'','Lunar Festival Execute sai if near npc 15905'),
+(22,4,180867,1,0,6,0,469,0,0,0,0,'','Lunar Festival SAI the player Must Be Alliance'),
+(22,5,180867,1,0,29,0,15908,5,0,0,0,'','Lunar Festival Execute sai if near npc 15908'),
+(22,5,180867,1,0,6,0,67,0,0,0,0,'','Lunar Festival SAI the player Must Must Be Horde'),
+(22,6,180867,1,0,29,0,15907,5,0,0,0,'','Lunar Festival Execute sai if near npc 15907'),
+(22,6,180867,1,0,6,0,67,0,0,0,0,'','Lunar Festival SAI the player Must Must Be Horde'),
+(22,7,180867,1,0,29,0,15719,5,0,0,0,'','Lunar Festival Execute sai if near npc 15719'),
+(22,7,180867,1,0,6,0,67,0,0,0,0,'','Lunar Festival SAI the player Must Must Be Horde');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1611685103659442481.sql:60]: newline

File: data/sql/updates/pending_db_world/rev_1611685103659442481.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+(26455, 1, -1031.45, -230.867, 160.14, 3.28358, 0);  -- To Thunderbluff
+
+DELETE FROM `spell_linked_spell` WHERE `spell_trigger`=26373 AND `spell_effect`=26448;
+DELETE FROM `conditions` WHERE `SourceEntry`=26373 AND `SourceTypeOrReferenceId`=13;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
+(13, 2, 26373, 0, 0, 31, 0, 4, 0, 0, 0, 0, '', 'Lunar Invitation visual on player');
+
+-- Lunar Festival spell focus for teleportaion
+UPDATE `gameobject_template` SET `AIName`="SmartGameObjectAI" WHERE `entry`=180867;
+DELETE FROM  `smart_scripts` WHERE `entryorguid` = 180867 AND source_type=1;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1611685103659442481.sql:30]: backticks

**Comment by pklloveyou** [comment]
Mark

**Comment by pklloveyou** [comment]
Tested

**Comment by FrancescoBorzi** [comment]
good job @Helias and thanks @pklloveyou for testing this!

**Pull request merged**